docs/src/main/sphinx/connector/snowflake.md

```diff
@@ -49,30 +49,163 @@ multiple instances of the Snowflake connector.
-| `time(n)`      | `time(n)`      |
-| `timestampntz` | `timestamp`    |
-
-Complete list of [Snowflake data types](https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html).
+Because Trino and Snowflake each support types that the other does not, this
+connector {ref}`modifies some types <type-mapping-overview>` when reading or
+writing data. Data types may not map the same way in both directions between
+Trino and the data source. Refer to the following sections for type mapping in
+each direction.
+
+List of [Snowflake data types](https://docs.snowflake.com/en/sql-reference/intro-summary-data-types.html).
+
+### Snowflake type to Trino type mapping
+
+The connector maps Snowflake types to the corresponding Trino types following
+this table:
+
+:::{list-table} Snowflake type to Trino type mapping
+:widths: 30, 30, 40
+:header-rows: 1
+
+* - Snowflake type
+  - Trino type
+  - Notes
+* - `NUMBER`
+  - `DECIMAL`
+  - Default precision and scale are (38,0).
+* - `DECIMAL`, `NUMERIC`
+  - `DECIMAL`
+  - Synonymous with `NUMBER`. See Snowflake
+    [data types for fixed point numbers](https://docs.snowflake.com/en/sql-reference/data-types-numeric#data-types-for-fixed-point-numbers)
+    for more information.
+* - `INT`, `INTEGER`, `BIGINT`, `SMALLINT`, `TINYINT`, `BYTEINT`
+  - `DECIMAL(38,0)`
+  - Synonymous with `NUMBER(38,0)`. See Snowflake
+    [data types for fixed point numbers](https://docs.snowflake.com/en/sql-reference/data-types-numeric#data-types-for-fixed-point-numbers)
+    for more information.
+* - `FLOAT`, `FLOAT4`, `FLOAT8`
+  - `DOUBLE`
+  - The names `FLOAT`, `FLOAT4`, and `FLOAT8` are for compatibility with other systems; Snowflake treats all three as
+    64-bit floating-point numbers. See Snowflake
+    [data types for floating point numbers](https://docs.snowflake.com/en/sql-reference/data-types-numeric#data-types-for-floating-point-numbers)
+    for more information.
+* - `DOUBLE`, `DOUBLE PRECISION`, `REAL`
+  - `DOUBLE`
+  - Synonymous with `FLOAT`. See Snowflake
+    [data types for floating point numbers](https://docs.snowflake.com/en/sql-reference/data-types-numeric#data-types-for-floating-point-numbers)
+    for more information.
+* - `VARCHAR`
+  - `VARCHAR`
+  -
+* - `CHAR`, `CHARACTER`
+  - `VARCHAR`
+  - Synonymous with `VARCHAR` except default length is `VARCHAR(1)`. See Snowflake
+    [String & Binary Data Types](https://docs.snowflake.com/en/sql-reference/data-types-text)
+    for more information.
+* - `STRING`, `TEXT`
+  - `VARCHAR`
+  - Synonymous with `VARCHAR`. See Snowflake
+    [String & Binary Data Types](https://docs.snowflake.com/en/sql-reference/data-types-text)
+    for more information.
+* - `BINARY`
+  - `VARBINARY`
+  -
+* - `VARBINARY`
+  - `VARBINARY`
+  - Synonymous with `BINARY`. See Snowflake
+    [String & Binary Data Types](https://docs.snowflake.com/en/sql-reference/data-types-text)
+    for more information.
+* - `BOOLEAN`
+  - `BOOLEAN`
+  -
```

Comment:
Move this type to top. We usually order as boolean → numeric → varchar/varbinary → temporal types → others.